pkg/controllers/dashboard/helm/repo.go

```diff
@@ -96,99 +118,28 @@ func (r *repoHandler) ClusterRepoDownloadStatusHandler(repo *catalog.ClusterRepo
 	})
 }
 
-func toOwnerObject(namespace string, owner metav1.OwnerReference) runtime.Object {
-	return &metav1.PartialObjectMetadata{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       owner.Kind,
-			APIVersion: owner.APIVersion,
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      owner.Name,
-			Namespace: namespace,
-			UID:       owner.UID,
-		},
-	}
-}
-
-func (r *repoHandler) createOrUpdateMap(namespace, name string, index *repo.IndexFile, owner metav1.OwnerReference) (*corev1.ConfigMap, error) {
-	// do this before we normalize the namespace
-	ownerObject := toOwnerObject(namespace, owner)
-
-	buf := &bytes.Buffer{}
-	gz := gzip.NewWriter(buf)
-	if err := json.NewEncoder(gz).Encode(index); err != nil {
-		return nil, err
-	}
-	if err := gz.Close(); err != nil {
-		return nil, err
-	}
-
-	if namespace == "" {
-		namespace = namespaces.System
-	}
-
-	var (
-		objs  []runtime.Object
-		bytes = buf.Bytes()
-		left  []byte
-		i     = 0
-		size  = len(bytes)
-	)
-
-	for {
-		if len(bytes) > maxSize {
-			left = bytes[maxSize:]
-			bytes = bytes[:maxSize]
-		}
-
-		next := ""
-		if len(left) > 0 {
-			next = name2.SafeConcatName(owner.Name, fmt.Sprint(i+1), string(owner.UID))
-		}
-
-		cm := &corev1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:            name2.SafeConcatName(owner.Name, fmt.Sprint(i), string(owner.UID)),
-				Namespace:       namespace,
-				OwnerReferences: []metav1.OwnerReference{owner},
-				Annotations: map[string]string{
-					"catalog.cattle.io/next": next,
-					// Size ensure the resource version should update even if this is the head of a multipart chunk
-					"catalog.cattle.io/size": fmt.Sprint(size),
-				},
-			},
-			BinaryData: map[string][]byte{
-				"content": bytes,
-			},
-		}
-
-		objs = append(objs, cm)
-		if len(left) == 0 {
-			break
-		}
-
-		i++
-		bytes = left
-		left = nil
-	}
-
-	return objs[0].(*corev1.ConfigMap), r.apply.WithOwner(ownerObject).ApplyObjects(objs...)
-}
-
+// The ensure method makes sure that a repo exists and is ready based on the provided RepoSpec
+// and RepoStatus. The repo's observed generation is set to the metadata's generation, and various
+// checks are made to determine whether the repo exists and is ready.
 func (r *repoHandler) ensure(repoSpec *catalog.RepoSpec, status catalog.RepoStatus, metadata *metav1.ObjectMeta) (catalog.RepoStatus, error) {
-	if status.Commit == "" {
+	// related ClusterRepo not opened or created yet
+	if status.Branch == "" || status.Branch != repoSpec.GitBranch {
```

Comment:
We should also check for the url ? if status.url != repoSpec.GitURl ? if there are not equal we should wait for download handler to run.